src/components/forms/Switch/Switch.tsx

```diff
@@ -1,70 +1,222 @@
-  line-height: ${pxToRem(30)};
+  display: inline-flex;
+  align-items: center;
+  height: ${getSwitchHeight};
+  line-height: ${getLineHeight('md')};
+  background: ${getFormStyle('switchBgColor')};
+  margin: 0;
+  border-radius: 16px;
+  cursor: pointer;
+  ${({ size }) => switchLabelWrapperSizes[size]};
+  ${({ isDisabled }) =>
+    css`
+      color: ${getFormStyle(isDisabled ? 'disabledColor' : 'switchColor')};
+      border: ${getFormStyle('borderWidth')} solid
+        ${getFormStyle(isDisabled ? 'disabledBorderColor' : 'borderColor')};
+    `};
+`;
+
+const LabelContent = styled.div<{
+  isDisabled?: boolean;
+  size?: Sizes;
+}>`
   font-size: ${getFontSize('md')};
-  color: ${getFormStyle('switchColor')};
-  transition: 0.3s;
+  font-family: ${getFontFamily('base')};
+  font-weight: ${getFontWeight('medium')};
+  cursor: pointer;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  ${({ size }) => switchNotCheckedLabelPaddings[size]};
 
-  &:after {
+  &::after {
     content: '';
     position: absolute;
-    top: ${pxToRem(4)};
-    left: ${pxToRem(5)};
-    width: ${pxToRem(20)};
-    height: ${pxToRem(20)};
-    background: ${getFormStyle('switchKnobBgColor')};
+    top: ${pxToRem(2)};
+    left: ${pxToRem(2)};
+    width: ${getSwitchLabelAfterElementSize};
+    height: ${getSwitchLabelAfterElementSize};
+    ${({ isDisabled }) =>
+      css`
+        background: ${getFormStyle(
+          isDisabled ? 'disabledColor' : 'switchKnobBgColor',
+        )};
+      `};
     border-radius: 20px;
     transition: 0.3s;
   }
+  transition: 0.3s;
 `;
 
-const Input = styled.input`
+const Input = styled.input<{
+  $size?: Sizes;
+}>`
   height: 0;
   width: 0;
   display: none;
-  &:checked + ${/* sc-selector */ Label} {
-    padding-left: ${pxToRem(15)};
-    padding-right: ${pxToRem(37)};
+  &:checked:disabled + ${/* sc-selector */ Label} {
+    color: ${getFormStyle('activeColor')};
+    background: ${getFormStyle('disabledColor')};
+    border-color: ${getFormStyle('disabledColor')};
+  }
+
+  &:checked + ${Label} {
+    ${LabelContent} {
+      ${({ $size }) => switchCheckedLabelPaddings[$size]};
+    }
+  }
+
+  &:checked + ${Label} {
     color: ${getFormStyle('activeColor')};
     background: ${getFormStyle('activeBorderColor')};
     border-color: ${getFormStyle('activeBorderColor')};
   }
-  &:checked + ${/* sc-selector */ Label}:after {
-    background: ${getFormStyle('activeColor')};
-    left: calc(100% - ${pxToRem(5)});
-    transform: translateX(-100%);
+  &:checked + ${Label} {
+    ${/* sc-selector */ LabelContent}::after {
+      background: ${getFormStyle('activeColor')};
+      left: calc(100% - ${pxToRem(2)});
+      transform: translateX(-100%);
+    }
   }
 `;
 
 const Switch: React.FC<
-  SwitchProps & React.PropsWithRef<JSX.IntrinsicElements['input']>
-> = ({ switchId, label, ...props }) => (
+  SwitchProps & React.PropsWithRef<Omit<JSX.IntrinsicElements['input'], 'size'>>
```

Comment:
@ajkl2533 regarding your comment on removing argTypes, I tried it before but it didn't work due to this line.

The comments on SwitchProps weren't being read. Apparently is because the React.propsWith.Ref...

I figured out if we add: `<ArgsTable of={Switch} />;` in Switch.stories.tsx, it works. Have you another better solution?

Reply:
Hm 🤔 not sure if this is correct typing. Why do you use `PropsWithRef`?
I would personally change it to:
```tsx
interface SwitchProps extends Omit<React.HTMLAttributes<HTMLInputElement>, 'size'> {
  ...
}

const Switch = forwardRef<
  HTMLInputElement,
  SwitchProps
>(
({ ... }, ref) => (...)
)
```
That way the descriptions in the `.types` file should be working, I'm using that in Stepper component